gradio/components/audio.py

```diff
@@ -17,13 +18,13 @@
 
 set_documentation_group("component")
 
-
-class WaveformOptions(TypedDict, total=False):
-    waveform_color: str
-    waveform_progress_color: str
-    show_controls: bool
-    skip_length: int
-
+@dataclasses.dataclass
```

Comment:
Thanks for fixing this @hannahblair. For backwards compatibility, it'd be good if we still accepted a `dict` of waveform options (and internally converted to this dataclass)

Reply:
Sorry I wasn't clear -- the dataclass for `WaveformOptions` you had before was great. But if someone has already written code where they pass `waveform_options` as a dictionary, we should continue to support that (by simply converting the dict to an object of type WaveformOptions`

e.g. something like:

```py
if isinstance(waveform_options, dict):
   waveform_options = WaveformOptions(**waveform_options)
```